### DISCUSSION TITLE
Dispatching events does not work

### DISCUSSION THREAD

**Discussion by wiwa1978** [livewire/livewire#8063] [1. Help]
I can't get events to work properly in my Laravel Livewire app.

- Laravel v10.46
- Livewire v3.4.6

The issue is that the event is never captured when dispatched. Here's the code:

```
//File: showProductDetail.php:

class showProductDetail extends Component { 
    public function viewProducts() {
        $this->dispatch('view-all-products')->to(ProductsByCategory::class); 
    }

    public function render() {
        return view('components.show-product-detail');
    }
}
```

```
//File: show-product-detail.blade.php:

<div>
   <button wire:click="viewProducts" type="button" class="text-sm">See all products</button>
</div>
```
```
//File: ProductsByCategory.php

class ProductsByCategory extends Component
{
    #[On('view-all-products')] 
    public function loadProducts() {
        dd('test');
    }
}
```
When I click the button I see (using developer tools) that an `update` is being transmitted. When I put a dd in the `viewProducts` function of `showProductDetail `, the dd fires (before and after the dispatch). 

Yet, the dispatched event does not seem to be captured by the `#[On('view-products')]` function, the `dd('test')` is not being executed.

I tried:

```
 $this->dispatch('view-products'); 
 $this->dispatch('view-products')->to(ProductsByCategory::class); 
```

Not sure if it matters but here's my app.blade.php

```
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>{{ $title ?? config('app.name', 'Laravel') }}</title>    
    @vite('resources/css/app.css')
    @livewireStyles
</head>

<body class="bg-white">
        @include('layouts.navigation')
        {{ $slot }}
    @livewireScripts
</body>
```

**Comment by danie-ramdhani** [comment]
works fine for me?

https://wirebox.app/b/xeg2k

**Reply by wiwa1978** [reply]
Apologies for late reply. I tried it out and indeed it works. It seems the trick is that you added `<livewire:products.products-by-category />` in the `simple.blade.php` for the event to trigger. If that is not there, the event would not trigger.

Ultimately what I would like to achieve is that if a user clicks on a product, the corresponding product detail page is loaded. I created a small repo to show that. I put your example for `Product` next to another model `Animal` (see [this](https://github.com/wiwa1978/LivewireEvents) repo). More in particular the `app/Livewire/Animals` and `app/Livewire/Animals/AnimalDetails.php` have the dispatch and the listener respectively. 

How can I use Livewire events to load the detail page for a specific animal? Or is it better to use plain old routes to achieve that?

**Reply by danie-ramdhani** [reply]
yes, if you want to display it better use redirect to another page not dispatch. dispatch is use to communicate between different components on the page (https://livewire.laravel.com/docs/events#dispatching-events).

i see you doing something like this:

```php
#[On('view-animal-details')]
public function render()
{
    return view('livewire.animals.animal-details');
}
```
it means you're refreshing the `AnimalDetails` component, not displaying it.

**Reply by wiwa1978** [reply]
Thanks again! Using a redirect would actually be a full-page reload. I'm somehow looking for a solution to avoid that.
